auth0/resource_auth0_connection_test.go

```diff
@@ -528,6 +528,50 @@ resource "auth0_connection" "google_oauth2" {
 }
 `
 
+func TestAccConnectionApple(t *testing.T) {
+
+	rand := random.String(6)
+
+	resource.Test(t, resource.TestCase{
+		Providers: map[string]terraform.ResourceProvider{
+			"auth0": Provider(),
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: random.Template(testAccConnectionAppleConfig, rand),
+				Check: resource.ComposeTestCheckFunc(
+					random.TestCheckResourceAttr("auth0_connection.apple", "name", "Acceptance-Test-Apple-{{.random}}", rand),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "strategy", "apple"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.client_id", "client_id"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.client_secret", "-----BEGIN PRIVATE KEY-----\nMIHBAgEAMA0GCSqGSIb3DQEBAQUABIGsMIGpAgEAAiEA3+luhVHxSJ8cv3VNzQDP\nEL6BPs7FjBq4oro0MWM+QJMCAwEAAQIgWbq6/pRK4/ZXV+ZTSj7zuxsWZuK5i3ET\nfR2TCEkZR3kCEQD2ElqDr/pY5aHA++9HioY9AhEA6PIxC1c/K3gJqu+K+EsfDwIQ\nG5MS8Y7Wzv9skOOqfKnZQQIQdG24vaZZ2GwiyOD5YKiLWQIQYNtrb3j0BWsT4LI+\nN9+l1g==\n-----END PRIVATE KEY-----"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.team_id", "team_id"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.key_id", "key_id"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.scopes.#", "2"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.scopes.2318696674", "name"),
+					resource.TestCheckResourceAttr("auth0_connection.apple", "options.0.scopes.881205744", "email"),
```

Comment:
Can we please add some update tests too? An additional step would be sufficient here to verify everything works as expected.

Reply:
Good idea, I'll add one now!

Reply:
Adding an update step seems to be tricky.

Something is off with the `scopes` when doing an update


The same thing happens with `testAccConnectionAzureADConfigUpdate` if you try to change the scopes in an update. For example, the following spec which removes a scope, will fail in a similar way:

```diff
diff --git a/auth0/resource_auth0_connection_test.go b/auth0/resource_auth0_connection_test.go
index e77cd86..3149edf 100644
--- a/auth0/resource_auth0_connection_test.go
+++ b/auth0/resource_auth0_connection_test.go
@@ -224,6 +224,14 @@ func TestAccConnectionAzureAD(t *testing.T) {
                                        resource.TestCheckResourceAttr("auth0_connection.azure_ad", "options.0.scopes.541325467", "ext_groups"),
                                ),
                        },
+                       {
+                               Config: random.Template(testAccConnectionAzureADConfigUpdate, rand),
+                               Check: resource.ComposeTestCheckFunc(
+                                       resource.TestCheckResourceAttr("auth0_connection.azure_ad", "options.0.scopes.#", "2"),
+                                       resource.TestCheckResourceAttr("auth0_connection.azure_ad", "options.0.scopes.1268340351", "ext_profile"),
+                                       resource.TestCheckResourceAttr("auth0_connection.azure_ad", "options.0.scopes.541325467", "ext_groups"),
+                               ),
+                       },
                },
        })
 }
@@ -255,6 +263,32 @@ resource "auth0_connection" "azure_ad" {
 }
 `

+const testAccConnectionAzureADConfigUpdate = `
+
+resource "auth0_connection" "azure_ad" {
+       name     = "Acceptance-Test-Azure-AD-{{.random}}"
+       strategy = "waad"
+       options {
+               client_id     = "123456"
+               client_secret = "123456"
+               tenant_domain = "example.onmicrosoft.com"
+               domain        = "example.onmicrosoft.com"
+               domain_aliases = [
+                       "example.com",
+                       "api.example.com"
+               ]
+               use_wsfed            = false
+               waad_protocol        = "openid-connect"
+               waad_common_endpoint = false
+               api_enable_users     = true
+               scopes               = [
+                       "ext_groups",
+                       "ext_profile"
+               ]
+       }
+}
+`
+
```

Reply:
Hmm, you are onto something there. But maybe we can make a separate issue to tackle this.

Perhaps write your update tests to keep the scopes and we can take care of the bug elsewhere.

Reply:
Yup, sounds reasonable. I'll try to get a working `update` test without testing the `scopes`  and we can tackle the `scopes` issue in another MR.

Reply:
Shall we move this to gitter? If you like I have a couple minutes https://gitter.im/terraform-provider-auth0/community